Feature request: Add configuration option to define JQL template #66 ...

lvasko opened the pull request:
...Added optional JQL template configuration option.
Also I corrected some old JavaDoc errors that my IDE was bitching about.
There is room for improvement, one may implement map of project keys and JQL templates, in case someone needs to have this functionality in bigger river-jira.
Tis functionality is quite handy for anyone who is working with JIRA over moderate size. In such cases projects are usually larger than normal and also querying those will impact usage for every user, as the Lucene index on which most queries are ran is the same for everyone.
Currently I have set up jqlTemplate parameter in a way that it defaults to exactly the same behavior as it was prior to this change.

velias commented:
Thanks a lot for the contribution. Can you please add unit test for jiraclient class which will set template by setter and then verify that it is correctly used in  prepareJIRAChangedIssuesJQL()?

lvasko commented:
I added this unittest to existing class and it started to interfer with another unittest "prepareJIRAChangedIssuesJQL()", so i added (re)setting default value also to that test.

velias commented:
Thanks a lot for the contribution. I released it as a new plugin version so you can use it.
